api_server/src/http_service.rs

```diff
@@ -520,28 +520,43 @@ impl hyper::server::Service for ApiServerHttpService {
                         // Requests on /mmds should not have the body in the logs as the data
                         // store contains customer data.
                         log_received_api_request(describe(&method_copy, &path, &None));
-                        let mut mmds = mmds_info
+                        let response = mmds_info
                             .lock()
-                            .expect("Failed to acquire lock on MMDS info");
-                        if mmds.is_initialized() {
-                            mmds.patch_data(json_value);
-                            Either::A(future::ok(empty_response(StatusCode::NoContent)))
-                        } else {
-                            Either::A(future::ok(json_response(
-                                StatusCode::NotFound,
-                                json_fault_message("The MMDS resource does not exist."),
-                            )))
+                            .expect("Failed to acquire lock on MMDS info")
+                            .patch_data(json_value);
+                        match response {
+                            Ok(_) => Either::A(future::ok(empty_response(StatusCode::NoContent))),
+                            Err(e) => match e {
+                                data_store::Error::NotFound => {
+                                    Either::A(future::ok(json_response(
+                                        StatusCode::NotFound,
+                                        json_fault_message(e.to_string()),
+                                    )))
+                                }
+                                data_store::Error::UnsupportedValueType => {
```

Comment:
Hi @andreeaflorescu ! Sorry to interrupt you. I may need a little help on creating tests to cover these lines I have added (which causes coverage failure). I couldn't find any existing unit tests that covers `ApiServerHttpService::call()`. Thanks for any suggestions!

Reply:
Sorry, nothing comes to mind right now. We could test `call` but it looks pretty daunting :(

Reply:
Thanks for your comments! I will add test for other untested code paths then.